From 7f10e4f93081fb52cb8ee2f76697358562576c20 Mon Sep 17 00:00:00 2001
From: Michael Adler <adlerm@mailbox.org>
Date: Sat, 10 Sep 2022 18:21:03 +0200
Subject: [PATCH 11/15] actualfullscreen

Actually toggle fullscreen for a window, instead of toggling the status
bar and the monocle layout.

https://dwm.suckless.org/patches/actualfullscreen/
---
 config.def.h | 1 +
 dwm.1        | 3 +++
 dwm.c        | 8 ++++++++
 3 files changed, 12 insertions(+)

diff --git a/config.def.h b/config.def.h
index ee9dc80..eaff8fc 100644
--- a/config.def.h
+++ b/config.def.h
@@ -113,6 +113,7 @@ static const Key keys[] = {
 	{ MODKEY|ControlMask,           XK_period, cyclelayout,    {.i = +1 } },
 	{ MODKEY,                       XK_space,  setlayout,      {0} },
 	{ MODKEY|ShiftMask,             XK_space,  togglefloating, {0} },
+	{ MODKEY|ShiftMask,             XK_f,      togglefullscr,  {0} },
 	{ MODKEY,                       XK_0,      view,           {.ui = ~0 } },
 	{ MODKEY|ShiftMask,             XK_0,      tag,            {.ui = ~0 } },
 	{ MODKEY,                       XK_comma,  focusmon,       {.i = -1 } },
diff --git a/dwm.1 b/dwm.1
index 829047b..31b23ec 100644
--- a/dwm.1
+++ b/dwm.1
@@ -122,6 +122,9 @@ Zooms/cycles focused window to/from master area (tiled layouts only).
 .B Mod1\-Shift\-c
 Close focused window.
 .TP
+.B Mod1\-Shift\-f
+Toggle fullscreen for focused window.
+.TP
 .B Mod1\-Shift\-space
 Toggle focused window between tiled and floating state.
 .TP
diff --git a/dwm.c b/dwm.c
index c8a1b42..f89af4d 100644
--- a/dwm.c
+++ b/dwm.c
@@ -249,6 +249,7 @@ static void tagmon(const Arg *arg);
 static void tile(Monitor *m);
 static void togglebar(const Arg *arg);
 static void togglefloating(const Arg *arg);
+static void togglefullscr(const Arg *arg);
 static void toggletag(const Arg *arg);
 static void toggleview(const Arg *arg);
 static void unfocus(Client *c, int setfocus);
@@ -1948,6 +1949,13 @@ togglefloating(const Arg *arg)
 	arrange(selmon);
 }
 
+void
+togglefullscr(const Arg *arg)
+{
+  if(selmon->sel)
+    setfullscreen(selmon->sel, !selmon->sel->isfullscreen);
+}
+
 void
 toggletag(const Arg *arg)
 {
-- 
2.38.1

